### PR TITLE
feat(installer): F.3 — carrier-merge file-carry via dir_overrides (w1qls.6.3)

### PR DESCRIPTION
**Pull request by scotthamilton77**
## F.3 — carrier-merge file-carry via `dir_overrides`

Implements story `agents-config-w1qls.6.3`. Completes the plugin carrier-merge
begun in F.2: F.2 validated the merge and cleared the `shared_carrier` flag but
did **not** represent the plugin's added files for sync. This PR adds the channel
that does.

### Stacked PR — retarget after #143
**This PR is stacked on #143** (`feat/w1qls.6.2-plugin-overlay`) and its base is
that branch, so the diff shows only F.3's changes. **After #143 merges, retarget
this PR's base to `main`** (and rebase if needed).

### What changed
- **`core/model.py`** — new `StagingPlan.dir_overrides: dict[Path, dict[Path, bytes]]`
  (`field(default_factory=dict)`; `StagingPlan` stays `@dataclass(slots=True)`).
  Keyed by the DIR item's `dest_relpath`, then by each inner file's relpath, to
  that file's bytes. Docstring documents both producers (carrier-merge now; F.5
  patched bytes later).
- **`core/overlay.py`** — on a successful carrier-merge, `_place` records the
  plugin DIR's disjoint files into `plan.dir_overrides[dest]` via the new
  `_carry_files` helper, then keeps the carrier item with the flag cleared (F.2
  behaviour unchanged). `_carry_files` mirrors the bash copy loop
  (`scripts/install.sh:585-592`): top-level dotfiles dropped (same exclusion as
  the disjoint check), subdirs carried wholesale (`cp -R` parity, including their
  nested dotfiles), keys are relpaths so a nested file lands at `subdir/file`.
- **`tests/unit/test_overlay.py`** — strengthened the carrier test to assert the
  plugin file's bytes are REPRESENTED in `dir_overrides`, plus edge cases:
  multiple plugin files, plugin-only payload (carrier's own files not duplicated),
  top-level-dotfile exclusion, nested-subdir carry, and the negative cases (plain
  disjoint add and no-collision skill overlay record nothing).

### Boundary — plan-level only
`core/sync.py` is still the single-file B.2 slice; the plan-walking DIR-sync that
would emit both the carrier's own tree and `dir_overrides[dest]` does **not**
exist yet and is **not** built here. Correctness is proven at the **plan layer**
(the override map is populated with the right bytes/keys). **Deployment proof is
deferred** to the future DIR-sync story.

### F.5 reuse note
`dir_overrides` is one primitive, two consumers. F.5's `apply_extensions` will
write YAML-patched bytes into this **same** channel for opaque skill/agent DIR
items (`StagedItem.content` stays `None`). The map is keyed by `dest_relpath` →
inner relpath; F.5 should write patched files under the same inner-relpath key
the carrier-merge uses (and be aware both producers may target the same `dest` —
last writer wins on a given inner relpath, which the future DIR-sync resolves by
overlaying `dir_overrides` on top of the source tree).

### Verification
`make ci-installer` green from the repo root: ruff lint + format, `mypy --strict`,
304 tests, **99.90%** coverage (≥90% floor), `pip-audit` clean, entry-verify pass.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: packages/installer/src/installer/core/overlay.py [hunk 1]
```diff
@@ -103,9 +103,12 @@ def _place(
...
-        # plugin's disjoint files conceptually merged in; clearing the flag
-        # makes any further plugin collision on this dir a true plugin-plugin
-        # collision (fatal via the registry).
+        # plugin's disjoint files merged in. Record those added files' bytes in
+        # the dir_overrides side channel (the carrier DIR item has a single
+        # source_path and so cannot itself express a second source tree), then
+        # clear the flag so any further plugin collision on this dir is a true
+        # plugin-plugin collision (fatal via the registry).
+        plan.dir_overrides[incoming.dest_relpath] = _carry_files(incoming.source_path)
         plan.items[incoming.dest_relpath] = replace(existing, shared_carrier=False)
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply packages/installer/src/installer/core/overlay.py:119]: Fixed in 6e569db. The carrier-merge now does `dir_overrides.setdefault(dest, {}).update(...)` instead of replacing the dest map, so contributions merge per inner relpath — last-writer-wins per file, not per directory. A prior producer (e.g. the later F.5 patched-bytes path) recording a file under a different inner relpath at the same dest now survives, with a test pinning that behavior.

File: packages/installer/src/installer/core/overlay.py [hunk 2]
```diff
@@ -138,3 +148,26 @@ def _visible_names(directory: Path) -> set[str]:
...
+    """The plugin DIR's disjoint files, keyed by their relpath under
+    ``directory`` to their bytes — the file-carry payload of a carrier-merge.
+
+    Mirrors the bash carrier-merge copy loop (``scripts/install.sh:585-592``):
+    ``for sfile in "$src"/*`` iterates the dot-excluded TOP-LEVEL entries, then
+    ``cp -R`` copies a subdir wholesale (its inner dotfiles included). So a
+    top-level dotfile is dropped — matching the same dotfile exclusion the
+    disjoint check applies via ``_visible_names`` — while a dotfile nested under
+    a carried subdir is kept. Keys are relpaths so a nested file lands at
+    ``subdir/file`` under the carrier DIR's destination."""
```
[the remark this answers is not in the document]
Comment (scotthamilton77) [reply packages/installer/src/installer/core/overlay.py]: Addressed in aa00197 by clarifying the docstring (option 2 in your comment). `_carry_files` records files only; an empty subdir is not representable in `dir_overrides`. That gap is intentional and unreachable here: plugin source trees are git-tracked and git cannot store an empty directory, so an empty subdir never reaches this function — an intentional empty dir ships a `.keep` placeholder, which is a file and is carried. Extending the plan model to represent directory entries would be dead weight for a case the source format cannot produce.
